byway/src/main/java/com/google/sps/servlets/GetTripsServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.sps.data.Trip;
+import com.google.sps.data.UserInfo;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/api/gettrips")
+public class GetTripsServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  @Override
+  public void init() {
+    System.setProperty(
+        DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserInfo userInfo = UserInfo.findOrCreateUser(userService, datastore);
+    if (userInfo == null) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+    }
+    List<String> userTripIds = userInfo.getTripIds();
+    ArrayList<Trip> userTrips = new ArrayList<Trip>();
+    for (String tripId : userTripIds) {
+      Trip trip = Trip.getTrip(datastore, tripId);
+      if (trip != null) {
+        userTrips.add(trip);
+      }
```

Comment:
Would be good to log a warning in the null case, so missing trips don't just get silently left out. That way you get the data that is usable, but also have something you can debug based on if it ever does happen.

Reply:
i added a log to the getTrips method in the Trip.java class to to log a warning for both the EntityNotFoundException and IllegalArgumentException